File: reverse_linkedlist/reverse_linkedlist.go
package reverselinkedlist

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	var reverseNode *ListNode

	current := head

	for current != nil {
		remainNode := current.Next
		current.Next = reverseNode
		reverseNode = current
		current = remainNode
	}

	return dummyNode
}

File: reverse_linkedlist/reverse_linkedlist.dart
class ListNode {
  int val;
  ListNode? next;
  ListNode([this.val = 0, this.next]);
}

class Solution {
  ListNode? reverseList(ListNode? head) {
    ListNode? reverseNode = null;
    var current = head;

    while (current != null) {
      var remainNode = current.next;
      current.next = reverseNode;
      reverseNode = current;
      current = remainNode;
    }

    return reverseNode;
  }
}
